lib/http/request.js

```diff
@@ -54,6 +57,35 @@ class HttpRequest extends EventEmitter {
     __globalSettings__ = val;
   }
 
+  static resetHttpKeepAliveAgents() {
+    __httpKeepAliveAgent__ = null;
+    __httpsKeepAliveAgent__ = null;
+  }
+
+  static getHttpKeepAliveAgent({keepAliveMsecs, maxSockets}) {
+    if (!__httpKeepAliveAgent__) {
```

Comment:
I was suggesting to refactor this into something like 

```js
function getAgent(secure) {
  const protocol = secure ? https : http;
  return __httpKeepAliveAgent__ = new protocol.Agent({
      keepAlive: true,
      keepAliveMsecs,
      maxSockets
    });
}
```

Reply:
Refactored and removed `__httpsKeepAliveAgent__` as we would need only a variable per test session/process (either `http` or `https` agent). For multiple envs we use worker_threads or child process which won't share instances of global variables.